packages/apidom-parser/src/parser.ts

```diff
@@ -18,119 +17,124 @@ export type {
 
 export { ParserError };
 
-const ApiDOMParser: stampit.Stamp<ApiDOMParserType> = stampit().init(
-  function ApiDOMParserConstructor() {
-    const adapters: ApiDOMParserAdapter[] = [];
+class ApiDOMParserClass implements ApiDOMParserType {
+  private adapters: ApiDOMParserAdapter[] = [];
 
-    const detectAdapterCandidates = async (source: string) => {
-      const candidates = [];
+  private async detectAdapterCandidates(source: string): Promise<ApiDOMParserAdapter[]> {
+    const candidates = [];
 
-      for (const adapter of adapters) {
-        // eslint-disable-next-line no-await-in-loop
-        if (isFunction(adapter.detect) && (await adapter.detect(source))) {
-          candidates.push(adapter);
-        }
+    for (const adapter of this.adapters) {
+      // eslint-disable-next-line no-await-in-loop
+      if (isFunction(adapter.detect) && (await adapter.detect(source))) {
+        candidates.push(adapter);
       }
+    }
 
-      return candidates;
-    };
+    return candidates;
+  }
 
-    const findAdapter = async (source: string, mediaType: string | undefined) => {
-      if (isString(mediaType)) {
-        return adapters.find((adapter) => {
-          if (!isArray(adapter.mediaTypes)) return false;
+  private async findAdapter(
+    source: string,
+    mediaType: string | undefined,
+  ): Promise<ApiDOMParserAdapter | undefined> {
+    if (isString(mediaType)) {
+      return this.adapters.find((adapter) => {
+        if (!isArray(adapter.mediaTypes)) return false;
 
-          return adapter.mediaTypes.includes(mediaType);
-        });
-      }
+        return adapter.mediaTypes.includes(mediaType);
+      });
+    }
 
-      const candidates = await detectAdapterCandidates(source);
+    const candidates = await this.detectAdapterCandidates(source);
 
-      return head(candidates);
-    };
+    return head(candidates);
+  }
 
-    this.use = function use(adapter: ApiDOMParserAdapter) {
-      adapters.push(adapter);
-      return this;
-    };
+  use(adapter: ApiDOMParserAdapter): this {
+    this.adapters.push(adapter);
+    return this;
+  }
 
-    this.findNamespace = async function findNamespace(
-      source: string,
-      options: ApiDOMParserOptions = {},
-    ) {
-      const adapter = await findAdapter(source, options.mediaType);
+  async findNamespace(
+    source: string,
+    options: ApiDOMParserOptions = {},
+  ): Promise<Namespace | undefined> {
+    const adapter = await this.findAdapter(source, options.mediaType);
 
-      return adapter?.namespace;
-    };
+    return adapter?.namespace;
+  }
 
-    this.findMediaType = async function findMediaType(source: string) {
-      const adapter = await findAdapter(source, undefined);
+  async findMediaType(source: string): Promise<string | void> {
```

Comment:
Can you maybe explain reason why you used `Promise<string | void>` instead of `Promise<string | undefined>`? 

Here is my understanding:

> Promise<string | undefined> is more explicit in indicating that the resolved value of the Promise can be either a string or undefined. On the other hand, Promise<string | void> suggests that the focus is not on the resolved value, which could be a string or nothing significant.